File: app/src/main/java/com/saint/struct/adapter/PagingAdapter.java
package com.saint.struct.adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.paging.PagingDataAdapter;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.saint.struct.R;
import com.saint.struct.bean.WanListBean;

public class PagingAdapter extends PagingDataAdapter<WanListBean, PagingAdapter.ViewHolder> {
    public PagingAdapter() {
        super(itemCallback);
    }

    @NonNull
    @Override
    public PagingAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_page_recycle, parent, false);
        return new PagingAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PagingAdapter.ViewHolder holder, int position) {
        WanListBean bean = getItem(position);
        if (bean != null) {
            holder.desc.setText(bean.desc);
            holder.date.setText(bean.niceDate);
            holder.title.setText(bean.title);
            holder.auth.setText(bean.author);
        }
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView desc;
        TextView date;
        TextView title;
        TextView auth;

        public ViewHolder(View itemView) {
            super(itemView);
            desc = itemView.findViewById(R.id.desc);
            date = itemView.findViewById(R.id.date);
            title = itemView.findViewById(R.id.title);
            auth = itemView.findViewById(R.id.author);
        }
    }

    private static DiffUtil.ItemCallback<WanListBean> itemCallback = new DiffUtil.ItemCallback<WanListBean>() {
        @Override
        public boolean areItemsTheSame(@NonNull WanListBean oldItem, @NonNull WanListBean newItem) {
            return oldItem.id.equals(newItem.id);
        }

        @SuppressLint("DiffUtilEquals")
        @Override
        public boolean areContentsTheSame(@NonNull WanListBean oldItem, @NonNull WanListBean newItem) {
            return oldItem.equals(newItem);
        }
    };
}

File: app/src/main/java/com/saint/struct/adapter/PagingAdapter.kt
package com.saint.struct.adapter

import androidx.paging.PagingDataAdapter
import com.saint.struct.bean.WanListBean
import android.view.ViewGroup
import android.view.LayoutInflater
import com.saint.struct.R
import android.annotation.SuppressLint
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView

class PagingAdapter : PagingDataAdapter<WanListBean, PagingAdapter.ViewHolder>(itemCallback) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_page_recycle, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val bean = getItem(position)
        if (bean != null) {
            holder.desc.text = bean.desc
            holder.date.text = bean.niceDate
            holder.title.text = bean.title
            holder.auth.text = bean.author
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var desc: TextView
        var date: TextView
        var title: TextView
        var auth: TextView

        init {
            desc = itemView.findViewById(R.id.desc)
            date = itemView.findViewById(R.id.date)
            title = itemView.findViewById(R.id.title)
            auth = itemView.findViewById(R.id.author)
        }
    }

    companion object {
        private val itemCallback: DiffUtil.ItemCallback<WanListBean> = object : DiffUtil.ItemCallback<WanListBean>() {
            override fun areItemsTheSame(oldItem: WanListBean, newItem: WanListBean): Boolean {
                return oldItem.id == newItem.id
            }

            @SuppressLint("DiffUtilEquals")
            override fun areContentsTheSame(oldItem: WanListBean, newItem: WanListBean): Boolean {
                return oldItem == newItem
            }
        }
    }
}